js/reference.js

```diff
@@ -1742,60 +1850,13 @@
-                            // dependent properties like filename, bytes_count_column, md5 and sha
-                            if (column.isAsset) {
-                                var isNull = newData[column.name] === null ? true : false;
-
-                                // If column has a filename column then add it to the projections
-                                if (column.filenameColumn) {
-                                    // If asset url is null then set filename also null
-                                    if (isNull) newData[column.filenameColumn.name] = null;
-                                    addProjection(column.filenameColumn.name);
-                                }
-
-                                // If column has a bytecount column thenadd it to the projections
-                                if (column.byteCountColumn) {
-                                    // If asset url is null then set filename also null
-                                    if (isNull) newData[column.byteCountColumn.name] = null;
-                                    addProjection(column.byteCountColumn.name);
-                                }
-
-                                // If column has a md5 column then add it to the projections
-                                if (column.md5 && typeof column.md5 === 'object') {
-                                    // If asset url is null then set filename also null
-                                    if (isNull) newData[column.md5.name] = null;
-                                    addProjection(column.md5.name);
-                                }
-
-                                // If column has a sha256 column then add it to the projections
-                                if (column.sha256 && typeof column.sha256 === 'object') {
-                                    // If asset url is null then set filename also null
-                                    if (isNull) newData[column.sha256.name] = null;
-                                    addProjection(column.sha256.name);
-                                }
-
-                                addProjection(column.name);
-
-                            } else {
-                                keyColumns = [];
-
-                                if (column.isKey) {
-                                    keyColumns = column.key.colset.columns;
-                                } else if (column.isForeignKey) {
-                                    keyColumns =  column.foreignKey.colset.columns;
-                                }
-
-                                for (n = 0; n < keyColumns.length; n++) {
-                                    keyColumnName = keyColumns[n].name;
-
-                                    addProjection(keyColumnName);
-                                }
+                        if (column.isInputIframe) {
+                            addProjectionForColumnObject(column);
```

Comment:
Looks like this case and the next case are where you are making changes to iterate over the other columns that are part of the input iframe annotation. I like how this is done so that as more properties are added/removed the code here doesn't need to change.

Could you add a comment here that states why this is done differently from the way asset annotation is done?

Should this be done similarly for asset columns that are part of an asset annotation? Currently each column is hardcoded in the above factored out functions.